pkg/kubelet/status/status_manager.go

```diff
@@ -685,12 +685,12 @@ type SidecarsStatus struct {
 func GetSidecarsStatus(pod *v1.Pod) SidecarsStatus {
 	var containerStatusesCopy []v1.ContainerStatus
 	if pod == nil {
-		klog.Infof("Pod was nil, returning empty sidecar status")
-		return SidecarsStatus{}
+		klog.Infof("Pod was nil, returning sidecar status that prevents progress")
+		return SidecarsStatus{SidecarsPresent: true, SidecarsReady: false, ContainersWaiting: true}
```

Comment:
If there is no pod, are the sidecar really present ?
I would assume that by default they're not , in the absence of spec. i would err on the side of caution.

Reply:
As discussed over chat, added to the function description comment to make it clear that we choose to be as defensive as possible and assume the worst in order to prevent `computePodActions()` from starting non-sidecars too early.